packages/components/order-tracking/src/order-tracking.controller.js

```diff
@@ -0,0 +1,143 @@
+
+export default class OrderTrackingController {
+  /* @ngInject */
+  constructor($log, $q, $timeout, OvhApiMeOrder) {
+    this.$log = $log;
+    this.$q = $q;
+    this.$timeout = $timeout;
+    this.OvhApiMeOrder = OvhApiMeOrder;
+    this.maxHistoryLength = ORDER_DEFAULT_HISTORY_LENGTH;
+  }
+
+  $onInit() {
+    return this.refresh();
+  }
+
+  refresh() {
+    this.order = null;
+    this.orderDetails = null;
+    this.orderFollowUp = null;
+    this.orderHistory = null;
+    this.polling = {
+      orderDetails: null,
+      orderFollowUp: null,
+    };
+    this.error = null;
+    this.isLoading = true;
+
+    this.OvhApiMeOrder.v6().resetAllCache();
+    this.cancelOrderDetailsPolling();
+    this.cancelOrderFollowUpPolling();
+
+    return this.OvhApiMeOrder.v6().get({
+      orderId: this.orderId,
+    }).$promise.then(order => this.getOrderStatus(order).then(({ status }) => {
+      const orderWithStatus = order;
+      orderWithStatus.status = status;
+      return orderWithStatus;
+    })).then((order) => {
+      this.order = order;
+      this.pollOrderDetails(0);
+      this.pollOrderFollowUp(0);
+    }).catch((error) => {
+      this.error = {
+        message: (error.data || { message: error.statusText }).message,
+      };
+      if (angular.isFunction(error.headers)) {
+        this.error.queryId = error.headers('x-ovh-queryid');
+      }
+    }).finally(() => {
+      this.isLoading = false;
+    });
+  }
+
+  $onDestroy() {
+    this.cancelOrderDetailsPolling();
+    this.cancelOrderFollowUpPolling();
+  }
+
+  getOrderStatus() {
+    return this.OvhApiMeOrder.v6().getStatus({
+      orderId: this.orderId,
+    }).$promise;
+  }
+
+  getOrderFollowUp() {
+    return this.OvhApiMeOrder.v6().followUp({
+      orderId: this.orderId,
+    }).$promise;
+  }
+
+  getOrderDetails() {
+    return this.OvhApiMeOrder.v6().getDetails({
+      orderId: this.orderId,
+    }).$promise.then(details => this.$q.all(details.map(id => this.OvhApiMeOrder.v6().getDetail({
+      orderId: this.orderId,
+      detailId: id,
+    }).$promise)));
+  }
+
+  pollOrderFollowUp(interval = ORDER_FOLLOW_UP_POLLING_INTERVAL) {
+    this.polling.orderFollowUp = this.$timeout(() => {
+      this.OvhApiMeOrder.v6().resetAllCache();
+      this.getOrderFollowUp().then((followUp) => {
+        this.orderFollowUp = followUp;
+        this.orderHistory = reverse(flatten(map(followUp, follow => reverse(follow.history))));
+        if (this.order.status === 'notPaid' && this.orderHistory.length === 0) {
+          this.orderHistory.push({
+            date: this.order.date,
+            label: 'custom_payment_waiting',
+          });
+        }
+      }).catch((err) => {
+        // 404 is returned for older orders that does not contains followUp and history
+        if (err.status === 404) {
+          this.orderFollowUp = {};
+          this.orderHistory = [];
+        } else {
+          this.$log.error(err);
```

Comment:
Maybe a message for customer would be nicer than a log, don't you think ?

Reply:
since it's a polling error i was thinking it would be better to make it silent

Reply:
yes you're probably right, the thing i like with $log is that you still notify that something unexpected happened, it's possible de define how to handle $log in the application that is using the module so it's interesting because it's possible to make it completely silent if you disable logging for example. The responsibility goes to the host of the module on how to handle errors